Feat: data-* attribute page for MathMLElement documentation

yashrajbharti opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. 🙌 -->

<!--
Add details below to help us review your pull request (PR).
Explain your changes and link to a related issue or pull request.
Your PR may be delayed or closed if you don't provide enough information.
-->

### Description

- Point to MathML reference
- Created `data-*` for MathML
<!-- ✍️ Summarize your changes in one or two sentences. -->

### Motivation

<!-- ❓ Why are you making these changes and how do they help readers? -->

### Additional details

<!-- 🔗 Link to release notes, browser docs, bug trackers, source control, or other resources. -->

### Related issues and pull requests

<!-- 🔨 If this fully resolves a GitHub issue, use "Fixes #123" -->
<!-- 👉 Highlight related pull requests using "Relates to #123" -->
<!-- ❗ If another pull request must be merged first, use "**Depends on:** #123" -->

<!-- 🔎 After submitting, the 'Checks' tab of your PR shows the build status. -->

yashrajbharti committed:
Fix data-* attribute links in MathMLElement.dataset documentation

- Change incorrect MathML reference paths to correct HTML paths
- Update both inline link and See also section
- data-* attributes are HTML global attributes, not MathML-specific

Josh-Cena commented:
Actually, perhaps you could update the parent page to also point to this hypothetical page as well?

yashrajbharti commented:
> Actually, perhaps you could update the parent page to also point to this hypothetical page as well?

Oh Understood!, doing that right now

yashrajbharti committed:
Update MathMLElement pages to reference MathML-specific data-* attributes

- Fix typo: 'Valuez' -> 'Value' in dataset property documentation
- Add reference to MathML data-* global attributes page in parent MathMLElement page
- Both pages now correctly point to the hypothetical MathML data-* page instead of HTML version
- Addresses reviewer feedback to create consistent MathML-specific documentation

yashrajbharti committed:
point to hypothetical page

yashrajbharti committed:
Delete .vscode/settings.json

yashrajbharti committed:
Merge branch 'main' into fix-data-attributes-links

yashrajbharti committed:
Simplify MathMLElement.dataset description to reference MathML data-* directly

- Remove HTML-specific wording and comparisons
- Make it clear that MathML defines its own data-* attributes normatively
- Follows the same pattern as SVG which has its own data-* page
- References MathML spec: https://w3c.github.io/mathml/#parsing_wellformed (MathMLDataAttributes)

yashrajbharti committed:
Add MathML data-* global attributes page

- Create comprehensive data-* documentation for MathML following SVG pattern
- Includes proper MathML-specific examples and references
- Links to MathMLElement.dataset property and DOM interface
- Follows XML naming rules and restrictions
- Addresses the missing page that MathMLElement.dataset was referencing
- Based on normative MathML spec: https://w3c.github.io/mathml/#parsing_wellformed

yashrajbharti commented:
> Actually, perhaps you could update the parent page to also point to this hypothetical page as well?

I copied SVG data-_star_ so we can have this unwritten page

fred-wang commented on files/en-us/web/mathml/reference/global_attributes/data-_star_/index.md:
```diff
@@ -0,0 +1,55 @@
+<math>
+  <msup data-formula="euler" data-value="-1">
+    <mi>e</mi>
+    <mi>iπ</mi>
```
This would be written as `<mrow><mi>i</mi> <mi>π</mi></mrow>` in MathML.

yashrajbharti replied:
Fixed, and also changed at another place!

yashrajbharti committed:
fix eq

fred-wang reviewed (commented):
OK thanks. This LGTM for the MathML part but I'll let someone else approve.

yashrajbharti commented:
Thanks @fred-wang , also @Josh-Cena  wonder if you could take a look at it again 🙂

Josh-Cena reviewed (approved):
Thanks, I think this is reasonable. Great addition
